controllers/storagecluster/odfinfoconfig.go

```diff
@@ -0,0 +1,293 @@
+			return reconcile.Result{}, fmt.Errorf("failed to delete key %v in odf-info %v: %v", odfInfoKeyName, odfInfoConfigMap.Name, err)
+		}
+	} else {
+		if err = r.Client.Delete(r.ctx, odfInfoConfigMap); err != nil && !errors.IsNotFound(err) {
+			r.Log.Error(err, "Failed to delete odf-info config map.", "ConfigMap", klog.KRef(odfInfoConfigMap.Namespace, odfInfoConfigMap.Name))
+			return reconcile.Result{}, fmt.Errorf("failed to delete odf-info %v: %v", odfInfoConfigMap.Name, err)
+		}
+	}
+	return reconcile.Result{}, nil
+}
+
+func getOdfInfoData(r *StorageClusterReconciler, storageCluster *ocsv1.StorageCluster, clients string) (string, error) {
+	odfVersion, err := getOdfVersion(r, storageCluster)
+	if err != nil {
+		return "", err
+	}
+	cephFSId, err := getCephFsid(r, storageCluster)
+	if err != nil {
+		return "", err
+	}
+
+	odfDeploymentType := odfDeploymentTypeExternal
+	isDROptimized := false
+	// Set isDROptmized to "false" in case of external clusters as we currently don't have to way to determine
+	// if external cluster OSDs are using bluestore-rdr
+	if !storageCluster.Spec.ExternalStorage.Enable {
+		odfDeploymentType = odfDeploymentTypeInternal
+		if isDROptimized, err = getIsDROptimized(r, storageCluster); err != nil {
+			return "", err
+
+		}
+	}
+	var storageSystemName string
+	if storageSystemName, err = getStorageSystemName(storageCluster); err != nil {
+		return "", err
+	}
+
+	connectedClients, err := getConnectedClients(r, storageCluster, clients)
+	if err != nil {
+		return "", err
+	}
+
+	data := OdfInfoData{
+		Version:           odfVersion,
+		DeploymentType:    odfDeploymentType,
+		IsDROptimized:     isDROptimized,
+		StorageSystemName: storageSystemName,
+		Clients:           connectedClients,
+		StorageCluster: infoStorageCluster{
+			NamespacedName:          client.ObjectKeyFromObject(storageCluster),
+			StorageProviderEndpoint: storageCluster.Status.StorageProviderEndpoint,
+			CephClusterFSID:         cephFSId,
+		},
+	}
+	yamlData, err := yaml.Marshal(data)
+	if err != nil {
+		return "", err
+	}
+	return string(yamlData), nil
+
+}
+
+func getConnectedClients(r *StorageClusterReconciler, storageCluster *ocsv1.StorageCluster, clients string) ([]connectedClient, error) {
+	storageConsumers := &v1alpha1.StorageConsumerList{}
+	err := r.Client.List(r.ctx, storageConsumers, client.InNamespace(storageCluster.Namespace))
+	if err != nil {
+		return nil, err
+	}
+	var connectedClients []connectedClient
+	if err = yaml.Unmarshal([]byte(clients), connectedClients); err != nil {
+		return nil, err
+	}
+
+	for storageConsumerID := range storageConsumers.Items {
+		storageConsumer := storageConsumers.Items[storageConsumerID]
+		clusterID := storageConsumer.Status.Client.ClusterID
+		namespacedName := storageConsumer.Status.Client.NamespacedName
+		var namespacedNameObj types.NamespacedName
+		if err = yaml.Unmarshal([]byte(namespacedName), namespacedNameObj); err != nil {
+			return nil, err
+		}
```

Comment:
I am not sure what you are unmarshaling here. The type is a string of the form `<namespace>/<name>` (https://github.com/red-hat-storage/ocs-operator/blob/main/api/v1alpha1/storageconsumer_types.go#L80)

Reply:
Yes, and my intention was to convert it to a `NamespacedName`. I couldn't find a util function that does that.

Reply:
Unmarshal will not do it for you as `<namespace>/<name>` is not in YAML format.
If there is no util that does that then just do 
```
parts := namespacedName.Split("/")
if len(parts) != 2 {
    return fmt.Errorf("Invalid namespace name %s", namespacedName)
}

namespacedNameObject := types.NamespacedName{
    Name := parts[2],
    Namespace := parts[1],
}
```